docs/core/extensions/logging.md

```diff
@@ -0,0 +1,515 @@
+  - Console
+  - Debug
+  - EventSource
+  - EventLog
+  - AzureAppServicesFile
+  - AzureAppServicesBlob
+  - ApplicationInsights
+
+### Set log level by command line, environment variables, and other configuration
+
+Log level can be set by any of the [configuration providers](configuration-providers.md).
+
+The following commands:
+
+- Set the environment key `Logging:LogLevel:Microsoft` to a value of `Information` on Windows.
+- Test the settings when using an app created with the .NET Worker service templates. The `dotnet run` command must be run in the project directory after using `set`.
+
+```cmd
+set Logging__LogLevel__Microsoft=Information
+dotnet run
+```
+
+The preceding environment setting:
+
+- Is only set in processes launched from the command window they were set in.
+- Isn't read by browsers launched with Visual Studio.
+
+The following [setx](/windows-server/administration/windows-commands/setx) command also sets the environment key and value on Windows. Unlike `set`, `setx` settings are persisted. The `/M` switch sets the variable in the system environment. If `/M` isn't used, a user environment variable is set.
+
+```cmd
+setx Logging__LogLevel__Microsoft=Information /M
+```
+
+On [Azure App Service](https://azure.microsoft.com/services/app-service/), select **New application setting** on the **Settings > Configuration** page. Azure App Service application settings are:
+
+- Encrypted at rest and transmitted over an encrypted channel.
+- Exposed as environment variables.
+
+For more information on setting .NET configuration values using environment variables, see [environment variables](configuration-providers.md#environment-variable-configuration-provider).
+
+## How filtering rules are applied
+
+When an <xref:Microsoft.Extensions.Logging.ILogger%601> object is created, the <xref:Microsoft.Extensions.Logging.ILoggerFactory> object selects a single rule per provider to apply to that logger. All messages written by an `ILogger` instance are filtered based on the selected rules. The most specific rule for each provider and category pair is selected from the available rules.
+
+The following algorithm is used for each provider when an `ILogger` is created for a given category:
+
+- Select all rules that match the provider or its alias. If no match is found, select all rules with an empty provider.
+- From the result of the preceding step, select rules with longest matching category prefix. If no match is found, select all rules that don't specify a category.
+- If multiple rules are selected, take the **last** one.
+- If no rules are selected, use `MinimumLevel`.
+
+## Logging output from dotnet run and Visual Studio
+
+Logs created with the [default logging providers](logging-providers.md) are displayed:
+
+- In Visual Studio
+  - In the Debug output window when debugging.
+- In the console window when the app is run with `dotnet run`.
```

Comment:
Here it's not clear what group "default providers" refers to -- those loaded by CreateDefaultBuilder or all of the built-in ones?  It's definitely not true of all the built-in ones, and I suspect it's only true of Debug and Console providers.